src/main/java/com/zaxxer/hikari/HikariConfig.java

```diff
@@ -42,27 +41,30 @@
 import com.zaxxer.hikari.metrics.MetricsTrackerFactory;
 import com.zaxxer.hikari.util.PropertyElf;
 
+import static java.util.concurrent.TimeUnit.MINUTES;
+import static java.util.concurrent.TimeUnit.SECONDS;
+
 import static com.zaxxer.hikari.util.UtilityElf.getNullIfEmpty;
 
 public class HikariConfig implements HikariConfigMXBean
 {
    private static final Logger LOGGER = LoggerFactory.getLogger(HikariConfig.class);
 
-   private static final long CONNECTION_TIMEOUT = TimeUnit.SECONDS.toMillis(30);
-   private static final long VALIDATION_TIMEOUT = TimeUnit.SECONDS.toMillis(5);
-   private static final long IDLE_TIMEOUT = TimeUnit.MINUTES.toMillis(10);
-   private static final long MAX_LIFETIME = TimeUnit.MINUTES.toMillis(30);
+   private static final int CONNECTION_TIMEOUT = (int) SECONDS.toMillis(30);
+   private static final int VALIDATION_TIMEOUT = (int) SECONDS.toMillis(5);
+   private static final int IDLE_TIMEOUT = (int) MINUTES.toMillis(10);
+   private static final int MAX_LIFETIME = (int) MINUTES.toMillis(30);
 
    private static final AtomicInteger POOL_NUMBER;
    private static boolean unitTest;
 
    // Properties changeable at runtime through the MBean
    //
-   private volatile long connectionTimeout;
-   private volatile long validationTimeout;
-   private volatile long idleTimeout;
-   private volatile long leakDetectionThreshold;
-   private volatile long maxLifetime;
+   private volatile int connectionTimeout;
+   private volatile int validationTimeout;
+   private volatile int idleTimeout;
+   private volatile int leakDetectionThreshold;
+   private volatile int maxLifetime;
    private volatile int maxPoolSize;
```

Comment:
While it might seem convenient, we cannot change these, it will break the ABI (application binary compatibility).  Users who have code like:
```java
HikariConfig config = new HikariConfig();
config.setConnectionTimeout(TimeUnit.SECONDS.toMillis(20));
```
Would now get a compilation failure because ``long`` cannot be downcast to an ``int``.